x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_object_type_definition.test.ts

```diff
@@ -73,11 +73,11 @@ it('correctly determines attribute properties', () => {
           'attr#2',
           { key: 'attr#4', dangerouslyExposeValue: true },
         ]),
-        attributesToExcludeFromAAD: new Set(['attr#3']),
+        attributesToIncludeInAAD: new Set(['attr#3', 'attr#4']),
       },
       {
         shouldBeEncrypted: [true, true, false, true],
-        shouldBeExcludedFromAAD: [true, true, true, true],
+        shouldBeIncludedInAAD: [false, false, true, false], // will not include attr#4 because it is to be encrypted
```

Comment:
Encrypted attributes are never used to construct AAD. See:
https://github.com/elastic/kibana/blob/e2a793d6af43abb716bdb748759fbf94a3cb3c7d/x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_object_type_definition.ts#L55

So if someone did include an encrypted attribute in the include list, it wouldn't matter. But, I see your point that we may just not want to allow this configuration in the first place. It would be more careful. I'll make this change. 👍 

I also realized that we may want to refactor the block that collects the AAD attributes.
Replace:
```
    const attributesAAD: Record<string, unknown> = {};
    for (const [attributeKey, attributeValue] of Object.entries(attributes)) {
      if (typeDefinition.shouldBeIncludedInAAD(attributeKey)) {
        attributesAAD[attributeKey] = attributeValue;
      }
    }
```
With:
```
    // Would need to move this logic into the typeDefinition and pass in the actual attributes
    if (typeDefinition.attributesToIncludeInAAD) {
      // Collect all attributes (both keys and values) that should contribute to AAD.
      for (const attributeKey of typeDefinition.attributesToIncludeInAAD) {
        if (!typeDefinition.shouldBeEncrypted(attributeKey) && attributes.has(attributeKey)) {
          attributesAAD[attributeKey] = attributes[attributeKey];
        }
      }
    }
```

This way we're always looping on a smaller N. What do you think?

Reply:
> Encrypted attributes are never used to construct AAD. See:

Yeah, both configuration are harmless even though they are invalid, I just meant that:

* `shouldBeEncrypted: [attr#1]` + `shouldBeExcludedFromAAD: [attr#1]` results in the attribute NOT being included in AAD, aligning with the expectations of the consumer providing such definition (consumer asked to **exclude**) :green_circle: 
* `shouldBeEncrypted: [attr#1]` + `shouldBeIncludedInAAD: [attr#1]` also results in the attribute NOT being included in AAD, but in this case it doesn't match the expectation (consumer asked to **include**) :red_circle: 

> But, I see your point that we may just not want to allow this configuration in the first place. It would be more careful. I'll make this change.

Thanks!

> This way we're always looping on a smaller N. What do you think?

Agreed (the only nit is `attributes.has(attributeKey)` --> `Object.hasOwn(attributes, attributeKey)`)!

Reply:
> Agreed (the only nit is attributes.has(attributeKey) --> Object.hasOwn(attributes, attributeKey))!

Thanks! This was just some quick pseudo code to convey the idea. I'll make this change as well. 🙂